apm-sniffer/optional-reporter-plugins/kafka-reporter-plugin/src/main/java/org/apache/skywalking/apm/agent/core/kafka/KafkaJVMMetricsSender.java

```diff
@@ -96,4 +94,11 @@ public void offer(final JVMMetric metric) {
             queue.offer(metric);
         }
     }
+
+    @Override
+    public void onStatusChanged(KafkaConnectionStatus status) {
+        if (status == KafkaConnectionStatus.CONNECTED) {
+            producer = ServiceManager.INSTANCE.findService(KafkaProducerManager.class).getProducer();
+        }
+    }
```

Comment:
Seem like to do not have retry logical?

Reply:
retry logical means that it will retry in bootstrap if KafkaProducer creates failure.  see this class named KafkaProducerManager, it has a schedule thread pool to retry until KafkaProducer created